var/spack/repos/builtin/packages/hiop/package.py

```diff
@@ -203,6 +203,9 @@ def cmake_args(self):
                 ]
             )
 
+        with when("+cuda @0.7.1:"):
+            args.extend([self.define_from_variant("HIOP_USE_RESOLVE", "cusolver_lu")])
```

Comment:
This is wrong. The context manager:
```
with when(...):
    ...
```
works at class level, and only for directives. What was this meant to do?

Reply:
```suggestion
        if self.spec.satisfies("+cuda @0.7.1:"):
            args.extend([self.define_from_variant("HIOP_USE_RESOLVE", "cusolver_lu")])
```

This is what we could change it to?

The idea is that `HIOP_USE_RESOLVE` is only an option in `@0.7.1:`, and then I think the `+cuda` requirement is redundant.

Now that I think about it, won't CMake just reference this as an unused variable in earlier versions? I assume that we can get away with lumping this into our other `args.extend` call if so.

Reply:
> Now that I think about it, won't CMake just reference this as an unused variable in earlier versions?

Yes, it should

Reply:
Okay I simplified and moved this as discussed.

Reply:
Changing back turned this into a no-op of a PR... I think the issue is that when building `~cuda`, the option got mangled and CMake threw a fit. Unless a maintainer of hiop can remember why / reproduce the issue, or if there are build failures that need addressing, then I think we can close the PR.